packages/amplify-graphql-function-transformer/src/graphql-function-transformer.ts

```diff
@@ -178,6 +193,51 @@ export class FunctionTransformer extends TransformerPluginBase {
   };
 }
 
+/**
+ * The response mapping template for 'RequestResponse' (default) and 'Event' invocation types differ.
+ * Use this to generate the appropriate response mapping template.
+ * @param config the {@link FunctionDirectiveConfiguration} for `@function` definition.
+ * @returns the response mapping template used by AppSync to handle responses from the Lambda function invocation.
+ */
+const responseMappingTemplate = (config: FunctionDirectiveConfiguration): string => {
```

Comment:
nit: split each template generator into a separate method. (future PR is ok)

Reply:
+1 to splitting this up into dedicated methods based on invocation type.

Reply:
Done. Addressed in https://github.com/aws-amplify/amplify-category-api/pull/2868